x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/container/use_container_metrics_table.ts

```diff
@@ -83,33 +83,76 @@ export function useContainerMetricsTable({
 
 function seriesToContainerNodeMetricsRow(series: MetricsExplorerSeries): ContainerNodeMetricsRow {
   if (series.rows.length === 0) {
-    return {
-      name: series.id,
-      uptime: null,
-      averageCpuUsagePercent: null,
-      averageMemoryUsageMegabytes: null,
-    };
+    return rowWithoutMetrics(series.id);
   }
 
-  let uptime: number = 0;
-  let averageCpuUsagePercent: number = 0;
-  let averageMemoryUsageMegabytes: number = 0;
-  series.rows.forEach((row) => {
-    const metricValues = unpackMetrics(row);
-    uptime += metricValues.uptime ?? 0;
-    averageCpuUsagePercent += metricValues.averageCpuUsagePercent ?? 0;
-    averageMemoryUsageMegabytes += metricValues.averageMemoryUsageMegabytes ?? 0;
+  return {
+    name: series.id,
+    ...calculateMetricAverages(series.rows),
+  };
+}
+
+function rowWithoutMetrics(name: string) {
+  return {
+    name,
+    uptime: null,
+    averageCpuUsagePercent: null,
+    averageMemoryUsageMegabytes: null,
+  };
+}
+
+function calculateMetricAverages(rows: MetricsExplorerRow[]) {
```

Comment:
For each series (host/container/pod), I get a set of rows (buckets), each "row" has a property for each metric field we asked for. If the field did not exist in the documents for that bucket, it becomes null. 

So I go through and unpack each "row", and collect each non-null value in a list.
Then, for each list that has some values in it, I get the average of those values.
Otherwise we return null which leads the table to display the `N/A` text.